12bits jpeg compilation

## Expected behavior and actual behavior.

When explicitly requesting 12-bit JPEG support with the `--with-jpeg12` configuration parameter (as the configuration of official GDAL compilation in various ubuntu distributions), the configuration summary always shows that jpeg12 in tiff is not supported, with the message `"Internal libjpeg12 has not the same ABI as libjpeg 8 bit. Disabling JPEG-in-TIFF 12 bit`".
The internal jpeg and tiff librairies are used (with_jpeg ="internal").

The problem comes from the "configure.ac" file, which performs a test to check the jpeg library version (JPEG_LIB_VERSION) by compiling a small "check_jpeg_abi.c" file.
For some reason, the result of compiling this file produces the following message: 
```
check_jpeg_abi.c:3:10: fatal error: jpeglib.h: No such file or directory
 #include "jpeglib.h"
          ^~~~~~~~~~~
compilation terminated.
```

The great specificity of this issue is that when the jpeg12 support is **not explicitly requested,** it is activated and works correctly!
Indeed, when the `--with-jpeg12` parameter is not mentioned, and when the internal jpeg and tiff libraries are used (as in most configurations), the compilation takes a way that no longer tests the jpeg version, and directly activates the 12 bit JPEG-in-TIFF support

## Steps to reproduce the problem.

./configure --with-jpeg12=yes --with-jpeg=internal

## Operating system

Ubuntu 16.04 64 bit

## GDAL version and provenance

Master from github

--- a/gdal/configure
+++ b/gdal/configure
@@ -25197,7 +25197,18 @@ fi
 TIFF_JPEG12_ENABLED=no
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for jpeg12" >&5
 $as_echo_n "checking for jpeg12... " >&6; }
-if test "$with_jpeg12" = yes ; then
+if test "$with_jpeg12" =  no ; then
+    JPEG12_ENABLED=no
+    { $as_echo "$as_me:${as_lineno-$LINENO}: result: disabled by user" >&5
+$as_echo "disabled by user" >&6; }
+
+elif test "$JPEG_SETTING" = "internal" -a "$TIFF_SETTING" = "internal" ; then
+    { $as_echo "$as_me:${as_lineno-$LINENO}: result: enabled" >&5
+$as_echo "enabled" >&6; }
+    JPEG12_ENABLED=yes
+    TIFF_JPEG12_ENABLED=yes
+
+elif test "$with_jpeg12" = yes ; then
     { $as_echo "$as_me:${as_lineno-$LINENO}: result: enabled" >&5
 $as_echo "enabled" >&6; }
     JPEG12_ENABLED=yes
@@ -25219,21 +25230,10 @@ $as_echo "$as_me: WARNING: Internal libjpeg12 has not the same ABI as libjpeg 8 
     fi
     rm -f check_jpeg_abi.*
 
-elif test x"$with_jpeg12" = x ; then
-    if test "$JPEG_SETTING" = "internal" -a "$TIFF_SETTING" = "internal" ; then
-      { $as_echo "$as_me:${as_lineno-$LINENO}: result: enabled" >&5
-$as_echo "enabled" >&6; }
-      JPEG12_ENABLED=yes
-      TIFF_JPEG12_ENABLED=yes
-    else
-      JPEG12_ENABLED=no
-      { $as_echo "$as_me:${as_lineno-$LINENO}: result: disabled, libjpeg or libtiff not internal" >&5
-$as_echo "disabled, libjpeg or libtiff not internal" >&6; }
-    fi
 else
     JPEG12_ENABLED=no
-    { $as_echo "$as_me:${as_lineno-$LINENO}: result: disabled by user" >&5
-$as_echo "disabled by user" >&6; }
+    { $as_echo "$as_me:${as_lineno-$LINENO}: result: disabled, libjpeg or libtiff not internal" >&5
+$as_echo "disabled, libjpeg or libtiff not internal" >&6; }
 fi
 
 JPEG12_ENABLED=$JPEG12_ENABLED

--- a/gdal/configure.ac
+++ b/gdal/configure.ac
@@ -1864,7 +1864,16 @@ AC_ARG_WITH([jpeg12],
 
 TIFF_JPEG12_ENABLED=no
 AC_MSG_CHECKING([for jpeg12])
-if test "$with_jpeg12" = yes ; then
+if test "$with_jpeg12" =  no ; then
+    JPEG12_ENABLED=no
+    AC_MSG_RESULT([disabled by user])
+
+elif test "$JPEG_SETTING" = "internal" -a "$TIFF_SETTING" = "internal" ; then
+    AC_MSG_RESULT([enabled])
+    JPEG12_ENABLED=yes
+    TIFF_JPEG12_ENABLED=yes
+
+elif test "$with_jpeg12" = yes ; then
     AC_MSG_RESULT([enabled])
     JPEG12_ENABLED=yes
 
@@ -1884,18 +1893,9 @@ if test "$with_jpeg12" = yes ; then
     fi
     rm -f check_jpeg_abi.*
 
-elif test x"$with_jpeg12" = x ; then
-    if test "$JPEG_SETTING" = "internal" -a "$TIFF_SETTING" = "internal" ; then
-      AC_MSG_RESULT([enabled])
-      JPEG12_ENABLED=yes
-      TIFF_JPEG12_ENABLED=yes
-    else
-      JPEG12_ENABLED=no
-      AC_MSG_RESULT([disabled, libjpeg or libtiff not internal])
-    fi
 else
     JPEG12_ENABLED=no
-    AC_MSG_RESULT([disabled by user])
+    AC_MSG_RESULT([disabled, libjpeg or libtiff not internal])
 fi
 
 AC_SUBST(JPEG12_ENABLED,$JPEG12_ENABLED)

